lib/common/common/src/mmap_hashmap.rs

```diff
@@ -194,18 +194,34 @@ impl<K: Key + ?Sized> MmapHashMap<K> {
     }
 
     pub fn keys(&self) -> impl Iterator<Item = &K> {
-        (0..self.keys_count()).filter_map(|i| {
-            let entry = self.get_entry(i).ok()?;
-            K::from_bytes(entry)
+        (0..self.keys_count()).filter_map(|i| match self.get_entry(i) {
+            Ok(entry) => K::from_bytes(entry),
+            Err(err) => {
+                debug_assert!(false, "Error reading entry for key {i}: {err}");
+                log::error!("Error reading entry for key {i}: {err}");
+                None
+            }
```

Comment:
These matches and logs are repetitive and make function definitions bulky. I suggest to keep them happy-path-centric and use the following approach instead:

```rust
let entry = self
    .get_entry(i)
    .tap_err(|e| log_or_panic!("Error reading entry for key {i}: {e}"))
    .ok()?;
```


[`tap_err`] is from the `tap` crate.
`log_or_panic!` could be a simple wrapper around `debug_assert!(); log::error!()` that could be added to the `common` crate (maybe alongside [`error_logging.rs`]).

[`tap_err`]: https://docs.rs/tap/1.0.1/tap/trait.TapFallible.html#tymethod.tap_err
[`error_logging.rs`]: https://github.com/qdrant/qdrant/blob/9fa86106e85d734765efe2c686a3ce975e53fb38/lib/segment/src/common/error_logging.rs

Reply:
Sure.